web/src/components/TaskContent.js

```diff
@@ -0,0 +1,43 @@
+import React from "react";
+import styled from "@emotion/styled";
+import Badge from "@leafygreen-ui/badge";
+
+export default function TaskContent({ task }) {
+  return (
+    <TaskDescription>
+      <TaskName>{task.name}</TaskName>
+      <Badge variant={getColorForStatus(task.status)}>
+        {getTaskStatusDisplayName(task.status)}
+      </Badge>
+    </TaskDescription>
+  );
+}
+
+const TaskDescription = styled.div`
+  display: flex;
+  width: 100%;
+`;
+const TaskName = styled.span`
+  flex-grow: 1;
+`;
+
+const getColorForStatus = (status) => {
+  switch (status) {
+    case "Open":
+      return "blue";
+    case "InProgress":
+      return "yellow";
+    case "Complete":
+      return "green";
+    default:
+      return "gray";
+  }
+};
+
+const getTaskStatusDisplayName = (status) => {
+  const nbsp = String.fromCharCode(160); // Non-breaking space so that a badge is always one line
+  if (status === "InProgress") {
+    return `In${nbsp}Progress`;
```

Comment:
Does the nbsp; do anything different than "In Progress" (with a space)?

Reply:
Yeah without nbsp the two words "in" and "progress" will flow to separate lines (I'll move the comment that explains this from line 38 to line 40)